data/actions/scripts/other/potions.lua

```diff
@@ -158,14 +158,14 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
 				end
 
 				player:say(string.format(potions.stringOfVocation, vocationString, potion.minLevel), TALKTYPE_MONSTER_SAY)
-				return false
+				return true
 			end
 		end
 
 		-- Condition exists after the check!
 		if player:getCondition(CONDITION_EXHAUST_HEAL) then
 			player:sendTextMessage(MESSAGE_STATUS_SMALL, Game.getReturnMessage(RETURNVALUE_YOUAREEXHAUSTED))
-			return true
+			return false
```

Comment:
Idk if you're aware that, when on use returns false it will send text message to player that he does cannot use that item, so it will overwrite the exhausted message.